controllers/cloudstackmachine_controller.go

```diff
@@ -223,11 +223,14 @@ func (r *CloudStackMachineReconciliationRunner) GetOrCreateVMInstance() (retRes
 		r.Recorder.Eventf(r.ReconciliationSubject, "Warning", "Creating", CSMachineCreationFailed, err.Error())
 	}
 	if err == nil && !controllerutil.ContainsFinalizer(r.ReconciliationSubject, infrav1.MachineFinalizer) { // Fetched or Created?
+		// adding a finalizer will require CPAC reconcile-delete trying to destroy associated VM through instanceID,
+		// so if err is not nil, which means it could not get associated VM through instanceID or name, we should not add finalizer to CloudStackMachine,
+		// otherwise, reconciler-delete will be stuck trying to wait instanceID to ba available.
```

Comment:
```suggestion
		// Adding a finalizer will make reconcile-delete try to destroy the associated VM through its instanceID.
		// If err is not nil, it means CAPC could not get an associated VM through instanceID or name, so we should not add a finalizer to this CloudStackMachine.
		// Otherwise, reconcile-delete will be stuck trying to wait for instanceID to be available.
```